curriculum/challenges/english/20-upcoming-python/learn-recursion-by-solving-the-tower-of-hanoi-puzzle/64ddd65848a12919d7e1c7d0.md

```diff
@@ -0,0 +1,68 @@
+---
+id: 64ddd65848a12919d7e1c7d0
+title: Step 29
+challengeType: 20
+dashedName: step-29
+---
+
+# --description--
+
+It's time to move some code from the `move` function to the `make_allowed_move` function. Move the code nested inside the outermost `if`, except the first `print` call, inside your new function. Pay close attention to the indentation.
+Don't forget to remove the `pass` keyword.
+
+# --hints--
+
+You should move the code nested inside the outermost `if`, except the first `print` call, inside your new function.
+
+```js
+({ test: () => {
+  assert.match(code, /def\s+make_allowed_move\(rod1, rod2\):\s+forward\s+=\s+False\s+if\s+not\s+rods\[\s*target\s*\]\s*:\s+forward\s*=\s*True\s+elif\s+rods\[\s*source\s*\]\s+and\s+rods\[\s*target\s*\]\s+and\s+rods\[\s*source\s*\]\[-1\]\s<\srods\[\s*target\s*\]\[\s*-\s*1\s*\]:\s+forward\s*=\s*True\s+if\s+forward\s*:\s+print\(\s*f'Moving\sdisk\s\{\s*rods\[\s*source\s*\]\[\s*-\s*1\s*\]\s*\}\sfrom\s\{\s*source\s*\}\sto\s\{\s*target\s*\}'\s*\)\s+rods\[\s*target\s*\]\.append\(\s*rods\[\s*source\s*\]\.pop\(\s*\)\s*\)\s+else\s*:\s+print\(\s*f'Moving\sdisk\s\{\s*rods\[\s*target\s*\]\[\s*-\s*1\s*\]\}\sfrom\s\{\s*target\s*\}\sto\s\{\s*source\s*\}'\s*\)\s+rods\[\s*source\s*\].append\(\s*rods\[\s*target\s*\].pop\(\s*\)\s*\)\s+#\sdisplay\sour\sprogress\s+\print\(\s*rods\s*\)/);  
+  }
+})
+```
```

Comment:
This should be clearer. When I read _"outermost `if`"_, I think of:

```py
            if forward:
                print(f'Moving disk {rods[source][-1]} from {source} to {target}')
                rods[target].append(rods[source].pop())
```

Not the `if` statement holding the above.

Reply:
The *first* `if` maybe?

Reply:
Yes, that makes more sense 👍